src/question_tree.rb

```diff
@@ -27,14 +25,17 @@ class QuestionTree < Tree::TreeNode
     def ask_questions
       filtered_each.with_index do |question, index|
         next unless ask_conditional_question?(question)
-        yield(question, index + 1)
+        progress = "(#{index + 1}/#{questions_length})"
+        yield create_question(question, progress)
       end
     end
 
     def questions_length
-      num = 0
-      filtered_each { |_q| num += 1 }
-      num
+      @questions_length ||= begin
+        num = 0
+        filtered_each { |_q| num += 1 }
+        num
```

Comment:
Rather than re-implementing `Enumerable#length` can we just use it, like this:

```ruby
def questions_length
  @questions_length ||= filtered_each.to_a.length
end
```
?

The tests pass when I make this change so I presume this works fine.

Reply:
Done